arbitrum-docs/stylus/stylus-quickstart.md

```diff
@@ -11,209 +11,242 @@ import PublicPreviewBannerPartial from '../partials/_public-preview-banner-parti
+`cargo stylus new` generates a starter template that implements a Rust version of the [Solidity `Counter` smart contract example](https://github.com/OffchainLabs/counter_contract/blob/master/contracts/Counter.sol). 
 
-Then, develop your Rust program normally and take advantage of all the features the [stylus-sdk](https://github.com/OffchainLabs/stylus-sdk-rs) has to offer.
+Alternatively, you can create a more barebone example with a Stylus entry point locally by running `cargo stylus new --minimal <YOUR_PROJECT_NAME>`. 
+Note that the `--minimal` option refers to a contract that handles simple "bytes in, bytes out" operations. Hence, the `cargo stylus export-abi` command (introduced later) won't work with contracts that use the entry point style because they don't enforce the Solidity ABI.
 
-### Checking your Stylus project is valid
+At this point, you can move on to the next step of this guide or develop your Rust smart contract. Feel free to use the [Stylus Rust SDK reference](./reference/overview) as a starting point.
 
-To check whether or not your program will successfully **deploy and activate** onchain, use the `cargo stylus check` subcommand:
+## Checking if your Stylus project is valid
 
-```
+By running the ' cargo stylus check ' command, you can check if your program can be successfully **deployed and activated** onchain.
+
+**Important:** Ensure you run Docker for this command to work correctly.
+
+```shell
 cargo stylus check
 ```
 
-This command will attempt to verify that your program can be deployed and activated onchain without requiring a transaction by specifying a JSON-RPC endpoint. See `cargo stylus check --help` for more options.
+`cargo stylus check` executes a dry-run on your project by compiling your contract to WASM and verifying if it can be deployed and activated onchain.
 
-If the command above fails, you'll see detailed information about why your WASM will be rejected:
 
-```
+If the command above fails, you'll see detailed information about why your contract would be rejected:
+
+```shell
 Reading WASM file at bad-export.wat
 Compressed WASM size: 55 B
-Stylus checks failed: program predeployment check failed when checking against
+Stylus checks failed: program pre-deployment check failed when checking against
 ARB_WASM_ADDRESS 0x0000…0071: (code: -32000, message: program activation failed: failed to parse program)
 
 Caused by:
     binary exports reserved symbol stylus_ink_left
 
 Location:
     prover/src/binary.rs:493:9, data: None)
-
 ```
 
-To read more about what counts as valid vs. invalid user WASM programs, see [VALID_WASM](https://github.com/OffchainLabs/cargo-stylus/blob/main/main/VALID_WASM.md). If your program succeeds, you'll see the following message:
+When a program fails the check, it means that there was a problem with the WASM binary generated by `cargo-stylus`.
```

Comment:
Flagging that `cargo-stylus` is used here. The term should be consistent throughout the document.